### PR TITLE
Feature/cansave methods

### PR DESCRIPTION
**Pull request by nyckoo**
- there is still some bug when saving succesfully signed person form 
- issue with DateTime variables in employee form (BirthDate, DateOfEmployment) - cannot be changed from initial settings after form is being saved, same as to update of employee - forces dates to be the ones set in form

### CODE REVIEW

File: LogisticApp/ViewModel/Forms/FormManager.cs [hunk 1]
```diff
@@ -57,7 +57,10 @@ public void addMediator(WindowMediator mediator)
 
         private bool canSave(object param)
         {
-            return true;
+            if (_selectedViewModel != null)
+                return _selectedViewModel.canSave();
+            else
+                return true;
```
Comment (mtyton) [inline LogisticApp/ViewModel/Forms/FormManager.cs:63]: should we return false here?

File: LogisticApp/ViewModel/Forms/EmployeeCreateFormViewModel.cs [hunk 2]
```diff
@@ -98,11 +98,24 @@ public override void loadData(BaseEntity entity)
...
         {
             Creator.createOrUpdate("employee");
         }
 
+        public override bool canSave()
+        {
+            if (_firstname == "" ||
+                _lastname == "" ||
+                _hourlyPayment < 0 ||
+                DateTime.Now - _birthDate < TimeSpan.FromDays(5840) ||
```
Comment (mtyton) [inline LogisticApp/ViewModel/Forms/EmployeeCreateFormViewModel.cs:111]: This number of days is some precise date? Won't it be better to check DateTime.Today.Addyears(-number_of_years)?

File: LogisticApp/ViewModel/Forms/BaseFormViewModel.cs [hunk 3]
```diff
@@ -45,5 +45,10 @@ public virtual void save()
             throw new NotImplementedException("Implement save method for each form!!");
         }
 
+        public virtual bool canSave()
```
Comment (mtyton) [inline LogisticApp/ViewModel/Forms/BaseFormViewModel.cs:48]: :+1:

File: LogisticApp/ViewModel/Forms/AddressFormViewModel.cs [hunk 4]
```diff
@@ -118,5 +118,19 @@ public override void save()
             Creator.createOrUpdate("address");
         }
 
+        public override bool canSave()
```
Comment (mtyton) [inline LogisticApp/ViewModel/Forms/AddressFormViewModel.cs:121]: Simplified example:
```
        public override bool canSave()
        {
            return (_city == "" || _country == "" || _street == "" || _postalCode == "" || _buildingNumber == 0)
        }
```
You can simply return this and it should work :smile:

**Review by mtyton — CHANGES_REQUESTED** [review]
Those canSave methods can be simplified as shown on the example.
About those errors they were fixed yesterday but if yhou had not merge main to this branch (I can see you didn't :smiley_cat: )
those fixes won't appear here. After merging this to main branch it all should work!

File: LogisticApp/ViewModel/Forms/EmployeeCreateFormViewModel.cs (hunk 2 above)
Comment (nyckoo) [reply LogisticApp/ViewModel/Forms/EmployeeCreateFormViewModel.cs:111]: yup, this is exactly 1st of january 2000

File: LogisticApp/ViewModel/Forms/FormManager.cs (hunk 1 above)
Comment (nyckoo) [reply LogisticApp/ViewModel/Forms/FormManager.cs:63]: > Those canSave methods can be simplified as shown on the example.
> About those errors they were fixed yesterday but if yhou had not merge main to this branch (I can see you didn't 😺 )
> those fixes won't appear here. After merging this to main branch it all should work!

That's strange, this time i surely was pulling main before applying changes, i created the whole new branch as well, the previous one was named "can_save_method".
Sorry for such a misconception :x

File: LogisticApp/ViewModel/Forms/EmployeeCreateFormViewModel.cs (hunk 2 above)
Comment (mtyton) [reply LogisticApp/ViewModel/Forms/EmployeeCreateFormViewModel.cs:111]: > Won't it be better to check DateTime.Today.Addyears(-number_of_years)?
How about this?

File: LogisticApp/ViewModel/Forms/FormManager.cs (hunk 1 above)
Comment (mtyton) [reply LogisticApp/ViewModel/Forms/FormManager.cs:63]: We will discuss this later, when I'm back home

**Review by nyckoo — COMMENTED** [review]
conditions now are more transparent and added some corrections as well

File: LogisticApp/ViewModel/Forms/AddressFormViewModel.cs (hunk 4 above)
Comment (mtyton) [reply LogisticApp/ViewModel/Forms/AddressFormViewModel.cs:121]: :heart:

**Review by mtyton — APPROVED** [review]

**Pull request merged**